app/bundles/CoreBundle/Form/Type/ConfigType.php

```diff
@@ -451,6 +451,35 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ]
         );
 
+        $builder->add(
+            'default_daterange_filter',
+            'choice',
+            [
+                'choices' => [
+                    ''          => 'mautic.core.daterange.0days',
+                    '-24 Hours' => 'mautic.core.daterange.1days',
+                    '-48 Hours' => 'mautic.core.daterange.2days',
+                    '-72 Hours' => 'mautic.core.daterange.3days',
+                    '-1 week'   => 'mautic.core.daterange.1weeks',
+                    '-24 weeks' => 'mautic.core.daterange.2weeks',
+                    '-3 weeks'  => 'mautic.core.daterange.3weeks',
+                    '-1 month'  => 'mautic.core.daterange.1month',
+                    '-2 months' => 'mautic.core.daterange.2month',
+                    '-3 months' => 'mautic.core.daterange.3month',
```

Comment:
`$this->translator` is injected in this class. Do you think we could use [pluralization](https://symfony.com/doc/current/components/translation/usage.html#pluralization) instead of these static options?

Also please go through every option and look at the charts how they look like. 2 and 3 days or 2 and 3 months will show a lot of empty space. We should have values that looks good. Also adding some years option would be great.

Reply:
### Options

I don't agree that those predefined selections are the best we can come up with. This is how 2 day back look like:

<img width="1171" alt="screenshot 2018-11-25 at 16 55 01" src="https://user-images.githubusercontent.com/1235442/48981220-0526e180-f0d3-11e8-9c1f-36ba38a23850.png">

How would you update it to look better? I get most of the options. Maybe it's only me, but I don't understand why would anyone want to use the **2 and 3 days option**. **24 weeks**? That's 5.52328 months. I could google only pregnancy related articles. Do you know of a business time range related 24 weeks? Why the **2 months option**? Will anyone use 2 months? Is there some business who is tracking last 2 months? Why not use for example 6 months instead? I think that would be used more.

### Pluralization

I linked the docs with examples. But I can specify the example for exactly your case if you need it.

*messages.ini*:
```ini
mautic.core.daterange.month="1 Month| %count% Months"
```

*ConfigType.php*:
```php
'-3 month'  => $translator->transChoice('mautic.core.daterange.month', 3),
```

With this change you'll have 3 new translation strings instead of 9. Our translators will appreciate it.

Let me know if it helps like this.

Reply:
1. removed the 2 and 3 day options
2. added pluralization
3. fixed -24 weeks to be 2 weeks (typo)